pkg/controllers/provisioningv2/rke2/plansecret/plansecret.go

```diff
@@ -0,0 +1,182 @@
+package plansecret
+
+import (
+	"bufio"
+	"bytes"
+	"context"
+	"crypto/sha256"
+	"encoding/hex"
+	"fmt"
+	"strings"
+
+	"github.com/rancher/rancher/pkg/controllers/provisioningv2/rke2"
+	capicontrollers "github.com/rancher/rancher/pkg/generated/controllers/cluster.x-k8s.io/v1beta1"
+	rkev1controllers "github.com/rancher/rancher/pkg/generated/controllers/rke.cattle.io/v1"
+	"github.com/rancher/rancher/pkg/provisioningv2/rke2/planner"
+	"github.com/rancher/rancher/pkg/wrangler"
+	corecontrollers "github.com/rancher/wrangler/pkg/generated/controllers/core/v1"
+	"github.com/sirupsen/logrus"
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/labels"
+)
+
+type handler struct {
+	secrets             corecontrollers.SecretClient
+	machinesCache       capicontrollers.MachineCache
+	etcdSnapshotsClient rkev1controllers.ETCDSnapshotClient
+	etcdSnapshotsCache  rkev1controllers.ETCDSnapshotCache
+}
+
+func Register(ctx context.Context, clients *wrangler.Context) {
+	h := handler{
+		secrets:             clients.Core.Secret(),
+		machinesCache:       clients.CAPI.Machine().Cache(),
+		etcdSnapshotsClient: clients.RKE.ETCDSnapshot(),
+		etcdSnapshotsCache:  clients.RKE.ETCDSnapshot().Cache(),
+	}
+	clients.Core.Secret().OnChange(ctx, "plan-status", h.OnChange)
+}
+
+func (h *handler) OnChange(key string, secret *corev1.Secret) (*corev1.Secret, error) {
+	if secret == nil || secret.Type != rke2.SecretTypeMachinePlan || len(secret.Data) == 0 {
+		return secret, nil
+	}
+
+	if len(secret.Data) == 0 {
```

Comment:
youre already making this check at line 42, probably can remove

Reply:
hah, thanks for catching this. I totally grafted onto this controller and renamed it and it looks like that slipped through. I'll remove the check now.